### PR TITLE
feat: add tqdm bar in FeedbackDataset.push_to_argilla

### LINKED ISSUE #3119: Add a `tqdm` bar in `FeedbackDataset.push_to_argilla`
# Description

There is not much transparency when pushing datasets to Argilla, as there's no progress bar that one can see to check that the dataset is being uploaded properly, while for `FeedbackDataset.push_to_huggingface` there is a built-in `tqdm` and some logging messages to track the progress.

# Solution

Add a `tqdm` bar to `FeedbackDataset.push_to_argilla` when looping over the records to upload, either if those are all the records in `records` or just the latest/new records in `__new_records`.

# Relevant lines

https://github.com/argilla-io/argilla/blob/fa165c254209889df98fda30990b0767fec046b5/src/argilla/client/feedback/dataset.py#L468-L473

https://github.com/argilla-io/argilla/blob/fa165c254209889df98fda30990b0767fec046b5/src/argilla/client/feedback/dataset.py#L552-L556

### PR DESCRIPTION
**Pull request by manulpatel**
# Description

Add a `tqdm ` progress bar to `FeedbackDataset.push_to_argilla` when looping over the records to upload, either if those are all the records in records or just the latest/new records in __new_records.

Closes #3119 

**Type of change**
- [x] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**
Manully ran a test.py to push a dataset to argilla and checked the appreance of `tqdm `bar in the terminal while adding records.

**Checklist**

- [ ] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/client/feedback/dataset.py
Comment (alvarobartt) [inline src/argilla/client/feedback/dataset.py]: Maybe a nice addition here is to add `desc`, something like

```suggestion
                        records=[record.dict() for record in tqdm(self.__new_records[i : i + PUSHING_BATCH_SIZE], desc="Pushing records to Argilla...")],
```

**Review by alvarobartt — CHANGES_REQUESTED** [review]
LGTM, just missing the description! 👍🏻

File: src/argilla/client/feedback/dataset.py
Comment (manulpatel) [reply src/argilla/client/feedback/dataset.py]: Sure! Noted and added in the next commit.

**Comment by alvarobartt** [comment]
Hi again @manulpatel thanks again for the great work! I think we can merge this PR right after #3234 since we're also fixing some bugs there, would you mind updating the `CHANGELOG.md` with a brief description of your changes an the URL pointing to this PR? Thanks!

**Comment by manulpatel** [comment]
Sure! Np! I need to add my changes under `1.11.0` in `CHANGELOG.md` in the same PR, correct?

**Comment by alvarobartt** [comment]
> Sure! Np! I need to add my changes under `1.11.0` in `CHANGELOG.md` in the same PR, correct?

Right! Under 1.11.0 as we'll be releasing it tomorrow!

File: src/argilla/client/feedback/dataset.py
Comment (gabrielmbmb) [inline src/argilla/client/feedback/dataset.py]: This won't work! `json.loads` expects either a `str` or `bytes`. Can you revert this change?

File: src/argilla/client/feedback/dataset.py
Comment (gabrielmbmb) [inline src/argilla/client/feedback/dataset.py:518]: I think that it would be good to pass the `disable` parameter to the `tqdm` function, so user can choose if they want to display the progress bar or not.

A `show_progress` parameter could be added to the `push_to_argilla` method and then pass the value of this param to tqdm `disable` param.

```python
    def push_to_argilla(self, name: Optional[str] = None, workspace: Optional[Union[str, rg.Workspace]] = None, show_progress: bool = False) -> None:
        ...
   ```

**Review by gabrielmbmb — CHANGES_REQUESTED** [review]
Thanks for the PR @manulpatel! I've left some comments :)

File: src/argilla/client/feedback/dataset.py
Comment (alvarobartt) [reply src/argilla/client/feedback/dataset.py]: Indeed that's already modified in the #3234 branch as mentioned below! But yes, that would run into conflicts once the other PR is merged into `develop`

**Comment by alvarobartt** [comment]
Also @manulpatel make sure to install the `pre-commit` to avoid the PR from failing due to that :smile: To do so you just need to install `pip install pre-commit` and from the root directory of the project run `pre-commit install`, then before every commit the `pre-commit` will run and fix the issues if applicable!

**Comment by manulpatel** [comment]
@alvarobartt So sorry. Actually I have installed the pre-commit hooks locally but the above commit I did directly from github editor, which didn't run the . In the next commits I will ensure this strictly.

**Comment by alvarobartt** [comment]
> @alvarobartt So sorry. Actually I have installed the pre-commit hooks locally but the above commit I did directly from github editor, which didn't run the . In the next commits I will ensure this strictly.

No worries at all @manulpatel 😄

File: CHANGELOG.md
Comment (alvarobartt) [inline CHANGELOG.md]:
```suggestion
- Added `tqdm` progress bar to `FeedbackDataset.push_to_argilla` when looping over the records to upload ([#3233](https://github.com/argilla-io/argilla/pull/3233)).
```

File: src/argilla/client/feedback/dataset.py
Comment (alvarobartt) [inline src/argilla/client/feedback/dataset.py:1]: You may need to resolve some merge conflicts here before merging 😄

**Review by alvarobartt — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (manulpatel) [reply CHANGELOG.md]: Oh typo.

File: src/argilla/client/feedback/dataset.py
Comment (manulpatel) [reply src/argilla/client/feedback/dataset.py:1]: Yep new learnings.

File: src/argilla/client/feedback/dataset.py
Comment (manulpatel) [reply src/argilla/client/feedback/dataset.py:518]: Thanks for the review. I have updated the PR.

File: src/argilla/client/feedback/dataset.py
Comment (alvarobartt) [inline src/argilla/client/feedback/dataset.py:479]: Do you mind also including the `show_progress` arg in the docstring under the `Args` section? Thanks

File: src/argilla/client/feedback/dataset.py
Comment (alvarobartt) [inline src/argilla/client/feedback/dataset.py]: We should add both the `desc` and the `disable=show_progress` args here too! 👍🏻

File: src/argilla/client/feedback/dataset.py
Comment (manulpatel) [reply src/argilla/client/feedback/dataset.py:479]: Sure would love to update the docs.

**Review by alvarobartt — APPROVED** [review]

**Review by gabrielmbmb — APPROVED** [review]

**Pull request merged**
